examples/ota-provider-app/esp32/main/BdxOtaSender.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
```

Comment:
This file shares a lot of code with the [common implementation](https://github.com/project-chip/connectedhomeip/blob/master/examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp). Maybe a better abstraction can be added to reduce duplicate code.

Currently the `BdxOtaSender` is a mix of two functionalities:

* Handle the events from BDX and drive the transfer.
* Fetch the OTA image data

The second functionality can have different implementation while the first remains the same. Can we separate the first part into a common class and place it under `examples/ota-provider-app/ota-provider-common` or even `src/app/`?

In the abstraction class we can add callbacks fetching OTA data. The `mOnBlockQueryCallback` and other callback members may be a good fit.

Reply:
@gjc13 There is currently a refactor/redesign underway (see https://github.com/project-chip/connectedhomeip/pull/11605/files). In an effort to not spend too much time on a partial refactor in this PR which would need to be reworked again, I had suggested for some of the code to duplicate so there is at least a working ESP32 requestor/provider.

Reply:
Since this examples will need to be reworked, resolving this issue.